File: common/src/main/java/com/pluxity/authentication/entity/RefreshToken.java
package com.pluxity.authentication.entity;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.TimeToLive;
import org.springframework.data.redis.core.index.Indexed;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@RedisHash("refresh_token")
public class RefreshToken {

    @Id private String username;

    @Indexed private String token;

    @TimeToLive private int timeToLive;

    public static RefreshToken of(String username, String token, int timeToLive) {
        return RefreshToken.builder()
                .username(username)
                .token(token)
                .timeToLive(timeToLive)
                .build();
    }
}

File: common/src/main/java/com/pluxity/authentication/entity/RefreshToken.kt
package com.pluxity.authentication.entity

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.TimeToLive
import org.springframework.data.redis.core.index.Indexed

@RedisHash("refresh_token")
data class RefreshToken(
    @Id val username: String,
    @Indexed val token: String,
    @TimeToLive val timeToLive: Int,
) {
    companion object {
        @JvmStatic
        fun of(
            username: String,
            token: String,
            timeToLive: Int,
        ): RefreshToken = RefreshToken(username, token, timeToLive)
    }
}
